transcoder/docs/playback.md

```diff
@@ -0,0 +1,178 @@
+
+Segments can be generated from videos as-is (direct playback), or transcoded. Kyoo supports both options. Transcoding is on the fly, slightly ahead of when a client is expected to request segments. Transcoding may be done one segment at a time, or in batches, which generally results in better transcoding performance. Once segments are transcoded, they are cached in the storage backend (filesystem, S3) for a user-configurable duration, and eventually removed when they have not been recently accessed. Cleanup is handled as a background job, and old segments may not be removed immediately.
+
+The transcoding service is designed to be highly available. When multiple transcoding service instances are deployed at once and configured properly, users should not notice when at least one service fails. This holds true even when the failed instance(s) were transcoding a video being actively played. This is because Kyoo supports _distributed, parallel transcoding_. The service can be configured so that a minimum number of transcoder instances will transcode the parts of the same video. When multiple instances transcode the same parts of the same video at the same time, only one has to succeed for each segments for transcoding to be successful.
+
+Because no two segments are guaranteed to come from the same transcoder instance, it is critical that all segments are entirely independent of each other, and do not overlap. "Parallel segments", or segments covering the same video and same time range that are produced by different instances, must always start with a [I-frame](https://en.wikipedia.org/wiki/Video_compression_picture_types). The start-finish time interval of parallel segments must also match up exactly, with no extra (or missing) frames. Additionally, for interoperability with direct playback, segments must line up exactly with keyframes in the source video. See [here](https://zoriya.dev/blogs/transcoder/) for more information.
+
+## Playback
+
+### Playlist requests
+```mermaid
+sequenceDiagram
+    participant cvp as Client video player
+    box Transcoder service (1..N)
+        participant api as Web API
+        participant jobs as Job worker
+    end
+    box Backend (HA)
+        participant db as Postgres
+        participant fs as Storage
+    end
+
+    cvp ->> api: Request playlist
+    api ->> db:  Get video metadata (segmentation times)
+    db -->> api: Return result
+    critical Metadata not available
+        api -->> api: Generate metadata
+        api -) db: Cache metadata
+    end
+    api ->> api: Generate playlist
+    api -) db: Create transcoding job for first k segments
+    api ->> cvp: Return video playlist
```

Comment:
```suggestion
    api ->> cvp: Return video playlist
    api ->> api: Generate playlist
    api -) db: Create transcoding job for first k segments
```

we do not wait for the first k segments to be ready to return the playlist, i think this change would highlight it.

Reply:
Just to clarify the `-)` arrow represents an async action, but yea I can reorder this.

Are you sure that you'd prefer "Return video playlist" before "Generate playlist"?